core/src/data/bigdecimal_ext.rs

```diff
@@ -11,13 +12,18 @@ impl BigDecimalExt for BigDecimal {
         self.is_integer()
             .then(|| bigdecimal::ToPrimitive::to_i8(self))?
     }
-
     fn to_i64(&self) -> Option<i64> {
         match self.is_integer() {
             true => bigdecimal::ToPrimitive::to_i64(self),
             false => None,
         }
     }
+    fn to_i128(&self) -> Option<i128> {
+        match self.is_integer() {
+            true => bigdecimal::ToPrimitive::to_i128(self),
+            false => None,
+        }
+    }
```

Comment:
This comment is only for sharing information, you don't need to do action now at least in this PR.
Current implementation code looks good enough.

fyi.
I tested using this code below and it is expected to pass but it failed with `ValueError::FailedToParseNumber`.
```rust
test!(  
    Ok(Payload::Insert(1)),
    &format!("INSERT INTO Item VALUES ({}, {})", i128::MAX, i128::MAX - 10)
);
```

And the reason was because of `bigdecimal::ToPrimitive::to_i128(self)`
https://docs.rs/bigdecimal/latest/bigdecimal/trait.ToPrimitive.html#method.to_i128
https://docs.rs/num-traits/0.2.15/src/num_traits/cast.rs.html#62-64
```rust
fn to_i128(&self) -> Option<i128> {
    self.to_i64().map(From::from)
}
```
That... is wrong implementation and
> The default implementation converts through to_i64(). Types implementing this trait should override this method if they can represent a greater range.

They also mentioned... quite... not promising.
We can consider to replace `to_i128` to our version (not to use `bigdecimal`'s) later.

Reply:
ok.  Thanks for looking into this, and documenting the issue.